machine.go

```diff
@@ -131,9 +131,6 @@ func (cfg *Config) Validate() error {
 
 // Machine is the main object for manipulating Firecracker microVMs
 type Machine struct {
-	// Metadata is the associated metadata that will be sent to the firecracker
-	// process
-	Metadata interface{}
```

Comment:
Alternatively, have we considered actually storing the user-provided metadata here? There could be use cases having it available in this struct. But considering that metadata can be used to pass sensitive data to the VM, it may also be reasonable for it to be write-only from the SDK's perspective. I just want to make sure we're fully aware of what we're doing before we make this change.

Reply:
Yea, I thought about it some but I can't really think of a good use case where it makes sense. For instance there are two method, `EnableMetadata` and `SetMetadata` where `EnableMetadata` will inject the metadata handler into the machine's handler stack so when `Start` is called it'll automatically call that API. Then you have `SetMetadata` which calls the API directly. I think with the two methods we cover what we need.

And the concerns with it only being writable is covered by the two operations. I think with that said, removing it makes the most sense.

Reply:
> Yea, I thought about it some but I can't really think of a good use case where it makes sense. For instance there are two method, `EnableMetadata` and `SetMetadata` where `EnableMetadata` will inject the metadata handler into the machine's handler stack so when `Start` is called it'll automatically call that API. Then you have `SetMetadata` which calls the API directly. I think with the two methods we cover what we need.

How so? None of those options facilitate the retrieval of the metadata from within the SDK, which was basically what my question was about.

Reply:
> How so? None of those options facilitate the retrieval of the metadata from within the SDK, which was basically what my question was about.

Ah, sorry. Yea, I think that makes sense, but shouldn't the `Metadata` field be added when we add `GET` operations to metadata? Rather than having it do nothing currently? That seems a little confusing. The SDK currently has no way of getting the metadata.

Reply:
GET from the API is probably the right place to grab this.